endpoint/endpoint.go

```diff
@@ -110,15 +112,15 @@ func (t Targets) IsLess(o Targets) bool {
 // Endpoint is a high-level way of a connection between a service and an IP
 type Endpoint struct {
 	// The hostname of the DNS record
-	DNSName string
+	DNSName string `json:"dnsName,omitempty"`
 	// The targets the DNS record points to
-	Targets Targets
+	Targets Targets `json:"targets,omitempty"`
 	// RecordType type of record, e.g. CNAME, A, TXT etc
-	RecordType string
+	RecordType string `json:"recordType,omitempty"`
 	// TTL for the record
-	RecordTTL TTL
+	RecordTTL TTL `json:"recordTTL,omitempty"`
 	// Labels stores labels defined for the Endpoint
-	Labels Labels
+	Labels Labels `json:"labels,omitempty"`
 }
```

Comment:
There's currently a  PR https://github.com/kubernetes-incubator/external-dns/pull/650 which includes a  ProviderSpecific field to set provider specific settings for endpoints. 

Could you add it additionally?

Reply:
Included.